cognite/client/data_classes/shared.py

```diff
@@ -47,3 +47,94 @@ def __init__(self, count: int = None, value: Union[int, str] = None, **kwargs):
         self.value = value
 
     value = CognitePropertyClassUtil.declare_property("value")
+
+
+class Geometry(dict):
+    """Represents the points, curves and surfaces in the coordinate space.
+
+    Args: type (str): The geometry type. One of 'Point', 'MultiPoint, 'LineString', 'MultiLineString', 'Polygon', or 'MultiPolygon'.
+          coordinates (List): An array of the coordinates of the geometry. The structure of the elements in this array is determined by the type of geometry.
```

Comment:
Umm, that's bit unclear to me what structure to use for let's say `MultiLineString` or anything else.

Reply:
Fair. I'll flesh out the documentation a bit, based on what I've written in `service-contracts`. 
@maciejmrowiec do you think it's worth it to go for the approach you suggested of building up separate projects from point -> array[point] etc. here as well?

Reply:
Did we reach on any consensus here?

Reply:
I've been trying various ways of doing this. in the `service-contracts` PR with hopes of drawing inspiration from that. Haven't found a solution I'm happy with there yet. 

While we wait for @maciejmrowiec 's input, what are your thoughts, @me-ydv-5? The options are to either just explain the different shapes a bit more in text (with a link to [this page](https://github.com/cognitedata/doc.cognitedata.com/pull/543) , or to construct `Point`, `LineString` etc. as objects of each other. The latter is naturally more work but I think it will look cleaner.

Reply:
Hmm, I think it's better to have python objects for each of the geojson objects (GO) given we derive one GO from other. Overall I agree it might cost some of your developer time but I guess that gets compensated with better user experience and early input validation (even though partially). :)

Reply:
Just got the time to look at this again. I like the cleanliness of deriving the GO from each other. However I worry that the usability will suffer, since developers can't just paste in a big array anymore (unless we add some sort of parsing and transformation).

Examples
With the current approach, a developer can construct a `GeoLocation` object with a `Polygon` like this:
```
    geometry = Geometry(type="Polygon", coordinates=[
        [[30, 10], [40, 40], [20, 40], [10, 20], [30, 10]]
    ])
    gl = GeoLocation(type="Feature", geometry=geometry, properties=dict())
```

With the suggested approach, the developer would have to construct each object individually, as they rely on each other:
```
p1 = Point(coordinates=[30, 10])
p2 = Point(coordinates=[40, 40])
... etc

ls = LineString(coordinates=[p1, p2, ...])
polygon = Polygon(coordinates=[ls])
```

So more steps. Depending on how this feature will actually be used, this might be fine. And we can add constructors to just take in the big array and then parse it accordingly. But that's even more work.

Reply:
Letting it go for the time being. Can be changed in the future based on reviews from users, etc.